packages/bpk-component-calendar/src/BpkCalendarGrid.tsx

```diff
@@ -110,15 +111,28 @@ class BpkCalendarGrid extends Component<Props, State> {
   constructor(props: Props) {
     super(props);
 
-    // We cache expensive calculations (and identities) in state
     this.state = {
       calendarMonthWeeks: getCalendarMonthWeeks(
         props.month,
         props.weekStartsOn,
+        // Do not run expensive date format calculations in the constructor
       ),
     };
   }
 
+  // Defer expensive calculations to this point to improve INP.
+  componentDidMount(): void {
+    deferCallback(() =>
+      this.setState({
```

Comment:
Question is a bit vague - can you clarify?

Reply:
This is not a new strategy I have adopted, the current version also uses state with the following comment: 

> // We cache expensive calculations (and identities) in state

Reply:
Ahh gotcha. 
- Having that call in the constructor avoids running the method on every render, that'd obviously be really bad for performance
- With that in mind the value would have to be stored on the class via some arbitrary property
- I think it may have always been relevant to use `this.state` to ensure React re-rendered when new props were provided (honestly that's a guess). 
- I believe it'd be more pertinent to use `this.state / this.setState` now given we quickly update the values and require a re-render after executing the deferred callback